src/unchecked-uninit.md

```diff
@@ -40,59 +91,32 @@ dropping the old value: [`write`], [`copy`], and [`copy_nonoverlapping`].
 It should go without saying that these functions, if misused, will cause serious
 havoc or just straight up Undefined Behavior. The only things that these
 functions *themselves* require is that the locations you want to read and write
-are allocated. However the ways writing arbitrary bits to arbitrary
-locations of memory can break things are basically uncountable!
-
-Putting this all together, we get the following:
-
-```rust
-use std::mem;
-use std::ptr;
-
-// size of the array is hard-coded but easy to change. This means we can't
-// use [a, b, c] syntax to initialize the array, though!
-const SIZE: usize = 10;
-
-let mut x: [Box<u32>; SIZE];
-
-unsafe {
-    // convince Rust that x is Totally Initialized
-    x = mem::uninitialized();
-    for i in 0..SIZE {
-        // very carefully overwrite each index without reading it
-        // NOTE: exception safety is not a concern; Box can't panic
-        ptr::write(&mut x[i], Box::new(i as u32));
-    }
-}
-
-println!("{:?}", x);
-```
+are allocated and properly aligned. However, the ways writing arbitrary bits to
+arbitrary locations of memory can break things are basically uncountable!
 
 It's worth noting that you don't need to worry about `ptr::write`-style
 shenanigans with types which don't implement `Drop` or contain `Drop` types,
-because Rust knows not to try to drop them. Similarly you should be able to
-assign to fields of partially initialized structs directly if those fields don't
-contain any `Drop` types.
+because Rust knows not to try to drop them. This is what we relied on in the
+above example. Similarly you should be able to assign to fields of partially
+initialized structs directly if those fields don't contain any `Drop` types.
```

Comment:
glad to see this is still legit lol

Reply:
The "should" is weird though, now that I think about it. Are you or are you not able to do this?

This is basically explaining rustc's static initialization tracking.

Reply:
Wait; this seems to be contradicting https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-a-struct-field-by-field ? What's an example of this working?

Reply:
True, that comment is not really applicable. You have to always initialize through a raw pointer.